ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/SourceConsumerGroupOffset.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.entity;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SourceConsumerGroupOffset {
+
+  private final int partition;
+  private final long logStartOffset;
+  private final long logEndOffset;
+  private final long consumerOffset;
+
+  @JsonCreator
+  public SourceConsumerGroupOffset(
+      @JsonProperty("partition") final int partition,
+      @JsonProperty("logStartOffset") final long logStartOffset,
+      @JsonProperty("logEndOffset") final long logEndOffset,
+      @JsonProperty("consumerOffset") final long consumerOffset
+  ) {
+    this.partition = partition;
+    this.logStartOffset = logStartOffset;
+    this.logEndOffset = logEndOffset;
+    this.consumerOffset = consumerOffset;
```

Comment:
nit: validate params that will be stored in object state; ensuring object does not get into an invalid state.

i.e. ensure none of these are negative:

```
Preconditions.checkArgument(partition <= 0, "invalid partition: " +. partition);
 // etc
```